cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_predict_dense_default_batch_impl.i

```diff
@@ -178,8 +178,103 @@ public:
 protected:
     services::Status predictByAllTrees(size_t nTreesTotal, size_t nClasses, const DimType & dim);
 
-    void predictByTrees(algorithmFPType * res, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x);
-    void predictByTreesVector(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x);
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    using Dispetcher_t = gbt::prediction::internal::PredictDispetcher<hasUnorderedFeatures, hasAnyMissing>;
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    void predictByTrees(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x,
+                        const Dispetcher_t<hasUnorderedFeatures, hasAnyMissing>& dispetcher);
+    template <bool hasUnorderedFeatures, bool hasAnyMissing>
+    void predictByTreesVector(algorithmFPType * val, size_t iFirstTree, size_t nTrees, size_t nClasses, const algorithmFPType * x,
+                              const Dispetcher_t<hasUnorderedFeatures, hasAnyMissing>& dispetcher);
+
+    template <bool val>
+    struct BooleanConstant {
+        typedef BooleanConstant<val> type;
+    };
+
+    inline void updateResult(algorithmFPType * res, algorithmFPType * val, size_t iRow, size_t i, size_t nClasses, BooleanConstant<true> dispetcher) {
+        res[iRow + i] = algorithmFPType(getMaxClass(val + i * nClasses, nClasses));
```

Comment:
Is the `algorithmFPType()` necessary? Wouldn't the assignment convert it anyways?

Reply:
Also typo: `dispetcher` -> `dispatcher`.

Reply:
> Is the `algorithmFPType()` necessary? Wouldn't the assignment convert it anyways?

you are right! Thanks.

> Also typo: `dispetcher` -> `dispatcher`.
done